### ISSUE TITLE
OpenCV functions that require null pointers does not work, such as the frequently used `min_max_loc`

### ISSUE THREAD

**Issue by fzyzcjy** [twistedfall/opencv-rust#285]
Hi thanks for the binding! However, OpenCV functions that require null pointers does not work, such as the frequently used `min_max_loc`.

A concrete example:

```
            let (mut min_val, mut max_val) = (0.,0.);
            let (mut min_loc, mut max_loc) = (Point::default(), Point::default());
            min_max_loc(
                &m,
                &mut min_val,
                &mut max_val,
                &mut min_loc,
                &mut max_loc,
                &no_array()?,
            )?;
```

When `m` is multi-channel, it errors and says: `called `Result::unwrap()` on an `Err` value: OpenCV(4.5.3) /tmp/opencv-20210728-84579-13worgs/opencv-4.5.3/modules/core/src/minmax.cpp:1495: error: (-215:Assertion failed) (cn == 1 && (_mask.empty() || _mask.type() == CV_8U)) || (cn > 1 && _mask.empty() && !minIdx && !maxIdx) in function 'minMaxIdx'  (code: -215)`.

This is probably because OpenCV expect min_loc and max_loc to be *null pointers* when input image is multi-channel. However, we have no way to provide a null pointer to `min_loc`...

A possible fix is to generate API such as `Option<&mut Point>` instead of `&mut Point`, and convert Option::None to nullptr.

**Comment by twistedfall** [comment]
Thanks for the report! The fix for that is currently in `rel` branch. I'll release it as soon as it passes CI.

**Comment by fzyzcjy** [comment]
Thank you for the quick fix!

**Comment by twistedfall** [comment]
Fixed in 0.56.0

**Comment by fzyzcjy** [comment]
:tada:
